### DISCUSSION TITLE
Github Container registry (ghcr.io) packages not appearing in webinterface

### DISCUSSION THREAD

**Discussion by christian-korneck** [community/community#25775] [Actions]
<p>Hello Github community,</p>
<p>I wanted to try out the new <a href="https://github.blog/2020-09-01-introducing-github-container-registry/" rel="nofollow noopener">Github Container registry</a> that was announced today.</p>
<p>With the Docker CLI I could successfully log in (using a personal access token) and push an image:</p>
<pre><code class="lang-auto">$ docker login ghcr.io
Username: my-github-username
Password *******************
$ docker pull hello-world:latest
$ docker tag hello-world:latest ghcr.io/my-github-username/hello-world:latest
$ docker push ghcr.io/my-github-username/hello-world:latest
</code></pre>
<p>This seems to have worked. The container image seems to be available at the registry:</p>
<pre><code class="lang-auto">docker manifest inspect ghcr.io/my-github-username/hello-world:latest
{
	"schemaVersion": 2,
...
</code></pre>
<p>But it doesn’t seem to be public:</p>
<pre><code class="lang-auto">$ docker logout
docker manifest inspect ghcr.io/my-github-username/hello-world:latest
denied: unauthenticated: User cannot be authenticated with the token provided.
</code></pre>
<p>I was now looking to modify the permissions. <a href="https://docs.github.com/en/packages/managing-container-images-with-github-container-registry/configuring-access-control-and-visibility-for-container-images" rel="nofollow noopener">According to the docs</a> I should be able to see the newly pushed container image on my github profile under <code>packages</code>.</p>
<p>But I’m only getting the default “getting started” screen there (as if I had zero packages):</p>
<p><img src="https://user-images.githubusercontent.com/6320032/181093212-a2242724-f769-455f-94b6-f74139f9b3dc.png" alt="image" data-base62-sha1="p8OoiivuXn6S6nQbnHcLhlbHFx3" width="451" height="243"></p>
<p>Am I doing anything wrong? Any help would be appreciated. Thanks in advance!</p>

**Comment by clarkbw** [comment, marked as answer]
<p>:wave: Thanks for trying it out!</p>
<p>Packages are initially published as private. You should be able to find them by looking for it under your user account <a href="https://github.com/my-github-username?tab=packages&amp;visibility=private">https://github.com/my-github-username?tab=packages&amp;visibility=private</a></p>
<p>Now that we’ve launched publicly we’ll be wiring up support for associated containers to show up along side the repository. Until then you’ll need to go to your org or user account view to find them.</p>
<p>You can also go directly to by name <a href="https://github.com/users/my-github-username/packages/container/hello-world/">https://github.com/users/my-github-username/packages/container/hello-world/</a></p>
<p>Here’s the URL scheme:</p>
<ul>
<li><a href="https://github.com/orgs/ORG/packages/container/CONTAINER_NAME">https://github.com/orgs/ORG/packages/container/CONTAINER_NAME</a></li>
<li><a href="https://github.com/users/ACCOUNT/packages/container/CONTAINER_NAME">https://github.com/users/ACCOUNT/packages/container/CONTAINER_NAME</a></li>
</ul>

**Comment by christian-korneck** [comment]
<p>thanks for the direct URL <a class="mention" href="/clarkbw">@clarkbw</a> , unfortunately the package (container image) doesn’t appear there either (also changing <code>visibility</code> = <code>all</code> is the same):</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/c7c5a9539aa002b8c8d20b4303a6cb927d07789f.png" data-download-href="/uploads/short-url/svgvgt2NxH2j6XVye5XOY0WFLHx.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/6320032/181093215-3177b6fd-8882-4cbb-9e8b-899772f51296.png" alt="image" data-base62-sha1="svgvgt2NxH2j6XVye5XOY0WFLHx" width="681" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/c/c7c5a9539aa002b8c8d20b4303a6cb927d07789f_2_681x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/c/c7c5a9539aa002b8c8d20b4303a6cb927d07789f_2_1021x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/c/c7c5a9539aa002b8c8d20b4303a6cb927d07789f_2_1362x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/c7c5a9539aa002b8c8d20b4303a6cb927d07789f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1620×1188 73.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>But I definitely have a container image on <code>ghcr.io</code> that I can pull with Docker (once I’m logged in with a personal access token):</p>
<pre><code class="lang-auto">$ docker pull ghcr.io/my-github-username/hello-world:latest
latest: Pulling from my-github-username/hello-world
Digest: sha256:90659bf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cbc042
Status: Downloaded newer image for ghcr.io/christian-korneck/hello-world:latest
ghcr.io/my-github-username/hello-world:latest
</code></pre>
<p>Any ideas? Thanks!</p>

**Comment by christian-korneck** [comment]
<p>quick addition: I can indeed access this URL directly:</p>
<pre><code class="lang-auto">https://github.com/users/my-github-username/packages/container/hello-world/
</code></pre>
<p>So manually assembling this URL with the container name seems to resolve my issue. Thanks for that <a class="mention" href="/clarkbw">@clarkbw</a> !</p>
<p>(However, I can’t access this page from anywhere else and even this page incorrectly states that there are <code>0 packages</code>).</p>
<p><img src="https://user-images.githubusercontent.com/6320032/181093219-11a326bf-a080-4a2c-b276-4c114eb89701.png" alt="image" data-base62-sha1="jAHSdGynTP7KtZdB2QvFJSyRjm8" width="530" height="362"></p>

**Comment by clarkbw** [comment]
<p>Excellent! The <strong>Edit Package</strong> drop down will have options for changing the package visibility and deleting versions. The switch to public should allow for authentication free access.</p>
<p>The other connections from the repo and search are being hooked up right now.  Hang tight and we’ll have those addressed soon.</p>

**Comment by den-is** [comment]
<p>same here :frowning:<br>
any updates on that?<br>
That makes migration from <strong><a href="http://docker.pkg.github.com" rel="noopener nofollow ugc">docker.pkg.github.com</a></strong> to <strong><a href="http://ghcr.io" rel="noopener nofollow ugc">ghcr.io</a></strong> painful for us.<br>
give us at least some API call example</p>

**Comment by paraboul** [comment]
<p>Having the same issue here. I can access the package by directly accessing the container URL but it doesn’t show up in my private packages neither in its related repository.</p>
<p>Also note that accessing it through :</p>
<p>This work :<br>
<a href="https://github.com/USERNAME/REPOSITORY_NAME/pkgs/container/hello-world/" class="onebox" target="_blank" rel="noopener nofollow ugc">https://github.com/USERNAME/REPOSITORY_NAME/pkgs/container/hello-world/</a></p>
<p>This triggers a 500 error :<br>
<a href="https://github.com/users/USERNAME/packages/container/hello-world/" class="onebox" target="_blank" rel="noopener nofollow ugc">https://github.com/users/USERNAME/packages/container/hello-world/</a></p>
<p><strong>Edit</strong>: It seems to show up after the first time the image was pulled. The 500 error is also gone. Looks like there is some kind of lazy loading happening where the package is not associated to the repository or account until it’s pulled from docker.</p>

**Comment by mattwelke** [comment]
<p>Wow, thanks for posting this. I didn’t realize this issue was still a thing. My team will likely be switching to GitHub for container images next year. This is good to know.</p>
